taca/element/Element_Runs.py

```diff
@@ -446,17 +446,301 @@ def rsync_successful(self):
+            # But for now we have to do this by ourselves in this hard way
+            project_runstats = get_project_runstats(sub_demux, demux_runmanifest)
+            # Read in IndexAssignment.csv
+            assigned_csv = os.path.join(self.run_dir, f"Demultiplexing_{sub_demux}", "IndexAssignment.csv")
+            if os.path.exists(assigned_csv):
+                with open(assigned_csv, 'r') as assigned_file:
+                    reader = csv.DictReader(assigned_file)
+                    index_assignment = [row for row in reader]
+                for sample in index_assignment:
+                    if sample['Lane'] in lanes:
+                        project_runstats_sample = [d for d in project_runstats if d['SampleName'] == sample['SampleName'] and d['Lane'] == sample['Lane'] and d['ExpectedSequence'] == sample['I1']+sample['I2']]
+                        sample['sub_demux_count'] = sub_demux
+                        sample['PercentMismatch'] = project_runstats_sample[0]['PercentMismatch']
+                        sample['PercentQ30'] = project_runstats_sample[0]['PercentQ30']
+                        sample['PercentQ40'] = project_runstats_sample[0]['PercentQ40']
+                        sample['QualityScoreMean'] = project_runstats_sample[0]['QualityScoreMean']
+                        aggregated_assigned_indexes.append(sample)
+            else:
+                logger.warning(f"No IndexAssignment.csv file found for sub-demultiplexing {sub_demux}.")
+        # Remove redundant rows for PhiX
+        aggregated_assigned_indexes_filtered = []
+        unique_phiX_combination = set()
+        for sample in aggregated_assigned_indexes:
+            if sample['SampleName'] == 'PhiX':
+                combination = (sample['I1'], sample['I2'], sample['Lane'])
+                if combination not in unique_phiX_combination:
+                    aggregated_assigned_indexes_filtered.append(sample)
+                    unique_phiX_combination.add(combination)
+            else:
+                aggregated_assigned_indexes_filtered.append(sample)
+        # Sort the list by Lane, SampleName and sub_demux_count
+        aggregated_assigned_indexes_filtered_sorted = sorted(aggregated_assigned_indexes_filtered, key=lambda x: (x['Lane'], x['SampleName'], x['sub_demux_count']))
+        # Fix new sample number based on SampleName and Lane
+        sample_count = 0
+        previous_samplename_lane = ('NA', 'NA')
+        for sample in aggregated_assigned_indexes_filtered_sorted:
+            if (sample['SampleName'], sample['Lane']) != previous_samplename_lane:
+                sample_count += 1
+                previous_samplename_lane = (sample['SampleName'], sample['Lane'])
+            sample['SampleNumber'] = sample_count
+        # Write to a new UnassignedSequences.csv file under demux_dir
+        aggregated_assigned_indexes_csv = os.path.join(self.run_dir, self.demux_dir, "IndexAssignment.csv")
```

Comment:
I wonder if we should instead use a different file name? Just to avoid confusion of what file is created by the demultiplexing software and the aggregation? Or would it never be confusing?

Reply:
I have no preference on the file name. You can decide whether a different one shall be used.

Reply:
Let's leave it as it is. 👍